### PR TITLE
Feature/iteration2

### LINKED ISSUE #2: Display individual Movie Component
When a user clicks a movie, that movie's details should be rendered

### LINKED ISSUE #4: Return to main view
User should have a way to return to the main view of all movies

### PR DESCRIPTION
**Pull request by yiranli624**
# Pull Request

#### What's this PR do?
Add chosen movie feature. When user click on any single movie, the page will only display the details of chosen movie. When user click return button, the page will display all movies again.

#### Where should the reviewer start?
App.js Line 8;
ChosenMovie.js Line 1;
ChosenMovie.scss Line 1;
MovieCard.js  Line 4;
Movies.js Line 4;

#### How should this be manually tested?
run ```npm start```

#### Any background context you want to provide?
Additional style needed, Asynchronous next

#### Screenshots (if appropriate):
Screen Shot 2020-12-01 at 10 19 57 PM

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,20 +1,46 @@
...
+  constructor() {
+    super();
     this.state = {
-      movies: movieData.movies
+      movies: movieData.movies,
+      chosenMovie: null
     }
   }
 
+  handleClick = (id) => {
```
Comment (coopterrones) [inline src/App.js:16]: Looks great!

File: src/App.js [hunk 2]
```diff
@@ -1,20 +1,46 @@
...
+    this.setState({
+      chosenMovie: null
+    })
+  }
+
   render() {
     return (
       <main className='App'>
-        <Movies movies={this.state.movies}/>
+        {this.state.chosenMovie &&
```
Comment (coopterrones) [inline src/App.js:33]: The way you did this is 🤯 .... so excited to learn new things from you.

File: src/components/ChosenMovie.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import '../styles/ChosenMovie.scss';
+
+const ChosenMovie = ({movie, displayAllMovies}) => {
+  let genres;
+  //this is temp condition before fetch real data
+  if (movie.genres) {
+    genres = movie.genres.map(genre => <p key={genre}>{genre} </p>) ;
```
Comment (coopterrones) [inline src/components/ChosenMovie.js:8]: Great using Key here.. I think this should suffice, but maybe we should ask to see if this is a good way of unique in this use case.. Also I know they are required for lists in React, but I'm curious to see what is the exact reasoning moving forward.

File: src/components/ChosenMovie.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        <section className='genre'>{genres}</section>
+        <section className='monetary-info'>
+          <p>Budget: ${movie.budget}</p>
+          <p>Revenue: ${movie.revenue}</p>
+        </section>
+        
+        
+        <p>Tagline: {movie.tagline}</p>
+      </section>
+      <button onClick={displayAllMovies}>Go Back</button>
```
Comment (coopterrones) [inline src/components/ChosenMovie.js:37]: Go back working great! Think we can style differently moving forward.

File: src/components/ChosenMovie.js (hunk 3 above)
Comment (yiranli624) [reply src/components/ChosenMovie.js:8]: for sure! Great quick thinking of the key value!

File: src/styles/ChosenMovie.scss [hunk 5]
```diff
@@ -0,0 +1,30 @@
+.backdrop-img {
```
Comment (coopterrones) [inline src/styles/ChosenMovie.scss:1]: Liking the layout for the info and picture! I want to definitely try to get this overlay working with a black gradient faded... Gonna be dope if we do it..

**Review by coopterrones — COMMENTED** [review]
Looks very solid! Will definitely be messing around with the styling for all our views, but very happy with our progress so far! Great work.. Thank you for going over your additions with me.

**Pull request merged**